testing-plugin/src/test/kotlin/arrow/meta/plugin/testing/ExampleTest.kt

```diff
@@ -2,47 +2,118 @@ package arrow.meta.plugin.testing
 
 import org.junit.Test
 
-class ExampleTest : CompilationTest {
+class ExampleTest {
 
-  //
-  // TODO: waiting for the arrow-annotations release which contains higherkind annotation
-  //    classpaths = listOf(classpathOf("arrow-annotations:x.x.x"))
-  //
+  @Test
+  fun `checks that compiles`() {
+    val compilerTest = CompilerTest(
+      code = {
+        """
+        | fun hello(): String =
+        |   "Hello world!"
+        | 
+        """.source
+      },
+      assert = {
+        listOf(compiles)
+      }
+    )
+    compilerTest.run(interpreter)
+  }
+
+  @Test
+  fun `check an expression evaluation`() {
+    val compilerTest = CompilerTest(
+      code = {
+        """
+        | fun hello(): String =
+        |   "Hello world!"
+        | 
+        """.source
+      },
+      assert = {
+        listOf(
+          "hello()".source.evalsTo("Hello world!"))
+      }
+    )
+    compilerTest.run(interpreter)
+  }
 
   @Test
-  fun `accepts dependencies and the generated meta file to check the compilation result`() {
-    """
-    | import arrow.higherkind
-    | 
-    | //metadebug
-    | 
-    | @higherkind
-    | class Id2<out A>(val value: A)
-    | 
-    | val x: Id2Of<Int> = Id2(1)
-    | 
-    """ withDependencies listOf("arrow-annotations:rr-meta-prototype-integration-SNAPSHOT") compilesTo """
-      | import arrow.higherkind
-      | 
-      | //meta: <date>
-      | 
-      | @arrow.synthetic class ForId2 private constructor() { companion object }
-      | @arrow.synthetic typealias Id2Of<A> = arrow.Kind<ForId2, A>
-      | @arrow.synthetic typealias Id2KindedJ<A> = arrow.HkJ<ForId2, A>
-      | @arrow.synthetic fun <A> Id2Of<A>.fix(): Id2<A> =
-      | this as Id2<A>
-      | @arrow.synthetic @higherkind /* empty? */class Id2 <out A> public constructor (val value: A) : Id2Of<A> {}
-      | 
-      | val x: Id2Of<Int> = Id2(1)
-      | 
-      """
+  fun `check that fails`() {
+    val compilerTest = CompilerTest(
+      code = {
+        """
+        | classsss Error
+        | 
+        """.source
+      },
+      assert = {
+        listOf(fails)
+      }
+    )
+    compilerTest.run(interpreter)
   }
 
   @Test
-  fun `emits error diagnostic when compilation fails`() {
-    """
-    | classsss Error
-    | 
-    """ emitErrorDiagnostic "Expecting a top level declaration"
+  fun `check that emits an error diagnostic when compilation fails`() {
+    val compilerTest = CompilerTest(
+      code = {
+        """
+        | classsss Error
+        | 
+        """.source
+      },
+      assert = {
+        listOf(
```

Comment:
Is there a case with more than one assert?. Most assertion are simple so there should be an additional function for `CompilerTest { }` that does not require returning a list of asserts

Reply:
Now there is just one case: `quoteOutputMatches` + `evalsTo`. 

Maybe we could create 2 tests in that case. I'm going to try it.

Reply:
Changed!